src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/tests/ModellingGroupTests.kt

```diff
@@ -36,6 +37,65 @@ class ModellingGroupTests : DatabaseTest()
         }
     }
 
+    @Test
+    fun `can get modelling group memberships for user`()
+    {
+        JooqContext().use {
+            it.addGroup("a", "description a")
+            it.addGroup("b", "description b")
+            it.addGroup("c", "description c")
+
+            setupTestUser()
+        }
+
+        val permissions = PermissionSet(
+                "*/can-login"
+        )
+        val roles = setOf(
+                ReifiedRole("member", Scope.Specific("modelling-group", "a")),
+                ReifiedRole("member", Scope.Specific("modelling-group", "b"))
+        )
+
+        val token = TestUserHelper().getTokenForTestUser(permissions, roles=roles)
+        val response = RequestHelper().get("/modelling-groups/user/memberships/", token)
+
+        val validator = JSONValidator()
+        validator.validateSuccess(response.text)
+        val groups  = JsonLoader.fromString(response.text)["data"]
+        assertThat(groups.isArray).isTrue()
+        assertThat((groups.count())).isEqualTo(2)
+
+        assertThat(groups[0]["id"].textValue()).isEqualTo("a")
+        assertThat(groups[0]["description"].textValue()).isEqualTo("description a")
+
+        assertThat(groups[1]["id"].textValue()).isEqualTo("b")
+        assertThat(groups[1]["description"].textValue()).isEqualTo("description b")
+    }
+
+    @Test
+    fun `can get empty modelling group memberships for user with no member roles`()
```

Comment:
I think this is more appropriately a controller test.

Reply:
Probably makes even more sense as a logic test now I've split that out